apps/web/ui/analytics/use-analytics-query.tsx

```
@@ -0,0 +1,132 @@
+      : false;
+
+  const selectedTab: EventType = useMemo(() => {
+    const event = searchParams.get("event");
+
+    return EVENT_TYPES.find((t) => t === event) ?? defaultEvent;
+  }, [searchParams.get("event"), defaultEvent]);
+
+  const queryString = useMemo(() => {
+    const availableFilterParams = VALID_ANALYTICS_FILTERS.reduce(
+      (acc, filter) => ({
+        ...acc,
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)